nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/composite/CompositeUtil.java

```diff
@@ -637,50 +637,41 @@ public Locale getLocale(MultivaluedMap<String,String> requestHeaders) {
      *
      * @param similarClass
      */
-    private void loadModelExtensionMetadata(Class<?> similarClass) {
-        BufferedReader reader = null;
+    private static void loadModelExtensionMetadata(Class<?> similarClass) {
         try {
             Enumeration<URL> urls = similarClass.getClassLoader().getResources("META-INF/restmodelextensions");
             while (urls.hasMoreElements()) {
                 URL url = urls.nextElement();
-                reader = new BufferedReader(new InputStreamReader(url.openStream()));
-                while (reader.ready()) {
-                    final String line = reader.readLine();
-                    if ((line == null) || line.isEmpty()) {
-                        continue;
-                    }
-                    if (line.charAt(0) != '#') {
-                        if (!line.contains(":")) {
-                            RestLogging.restLogger.log(Level.INFO,
-                                    RestLogging.INCORRECTLY_FORMATTED_ENTRY,
-                                    new String[]{
-                                        "META-INF/restmodelextensions",
-                                        line
-                                    });
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()))) {
+                    while (reader.ready()) {
+                        final String line = reader.readLine();
+                        if ((line == null) || line.isEmpty()) {
+                            continue;
                         }
-                        String[] entry = line.split(":");
-                        String base = entry[0];
-                        String ext = entry[1];
-                        List<String> list = modelExtensions.get(base);
-                        if (list == null) {
-                            list = new ArrayList<String>();
-                            modelExtensions.put(base, list);
+                        if (line.charAt(0) != '#') {
```

Comment:
Just out of curiosity, why use line.charAt(0) != '#' over line.startsWith('#')?

Reply:
Don't know the context here but I'd assume `#` is used for line comments so any line starting with `#` is effectivly ignored by this check.

Reply:
@cubastanley is your questions answered?

Reply:
I think the question is still the same. If line **starting with** "#" is supposed to be ignored, why not use **startsWith** method? :)

Reply:
Opsi, somehow missed the _over_ part when reading it. 

`charAt` is _cheaper_, maybe that is why it was chosen. Or just personal style. I would not see `startsWith` as _more readable_ and maybe the author felt the same so he was opting for the option that execution-wise does exactly what he wants to check, no less, no more, which means less surprises of some odd corner case possibly existing with all you could do with a `String`. Using `charAt` you could skip thinking if such a corner case exist as it is a very basic operation. 
We will never know - I think both are just fine for such a check in general. Sometimes context can be a reason to chose one over the other. Within a character by character parser `charAt` is more clear, within string processing `startsWith`.